### PR TITLE
Additional Functionality

### PR DESCRIPTION
**Pull request by twangboy**
Added the following functions:
- run_config
- test_config
- get_config
- get_config_status

### CODE REVIEW

**Comment by jtand** [comment]
Go Go Jenkins!

File: salt/modules/dsc.py [hunk 1]
```diff
@@ -194,3 +194,69 @@ def remove(name):
...
     no_ret = _pshell(cmd)
     return name not in list_modules()
+
+
+def run_config(path):
+    # Run the DSC Configuration
+    cmd = '$job = Start-DscConfiguration -Path {0};'.format(path)
+    cmd += 'Do{ } While ($job.State -notin \'Completed\', \'Failed\'); ' \
+           'return $job.State'
+    ret = __salt__['cmd.run'](cmd, shell='powershell', python_shell=True)
```
Comment (cachedout) [inline salt/modules/dsc.py]: With `python_shell=True` are you certain this is not vulnerable to shell injection attacks? We should ensure that a maliciously crafted argument could not be inserted here.

**Comment by cachedout** [comment]
@twangboy Please see the in-line comments before we can get this merged. Thanks.

File: salt/modules/dsc.py [hunk 2]
```diff
@@ -194,3 +194,69 @@ def remove(name):
...
+    try:
+        ret = json.loads(ret, strict=False)
+    except ValueError as esc:
+        log.debug('Json not returned')
+        return ret
+
+    return ret
+
+
+def create_config(name, resource, **kwargs):
```
Comment (twangboy) [inline salt/modules/dsc.py]: Depending on the resource you use there will be different arguments. For example the User resource accepts UserName, FullName, Password, etc... While the Group resource accepts GroupName, Description, Members, etc...

File: salt/modules/dsc.py (hunk 1 above)
Comment (twangboy) [inline salt/modules/dsc.py]: This isn't meant to be production yet. Perhaps I shouldn't push this up to develop then?

File: salt/modules/dsc.py (hunk 2 above)
Comment (cachedout) [inline salt/modules/dsc.py]: I understand that, but can't you simply accept a kwarg for a single dictionary called 'options' or whatever name you like and iterate off of that? Given the code below, it's not clear why you would need to accept an indeterminate number of kwargs, when a single dictionary would do.

File: salt/modules/dsc.py (hunk 1 above)
Comment (cachedout) [inline salt/modules/dsc.py]: We certainly accept non-production-ready code into the develop branch, which has never been guaranteed to be stable. That said, it's generally a good practice to document experimental code as-such to ensure that it isn't accidentally used by anybody and to create an issue to track the readiness of the code. Obviously, the last thing we want to do is to cut a release with code in that is not ready to go.

**Pull request closed without merging**